code/modules/mob/living/carbon/carbon.dm

```diff
@@ -591,17 +591,19 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 				hurt = FALSE*/
 	if(hit_atom.density && isturf(hit_atom))
 		if(hurt)
-			Weaken(2 SECONDS)
-			take_organ_damage(10)
+			visible_message("<span class='danger'>[src] slams into [hit_atom]!</span>", "<span class='userdanger'>You slam into [hit_atom]!</span>")
+			take_organ_damage(7.5 * speed) // speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
+			KnockDown(2 SECONDS)
+			playsound(src, 'sound/weapons/punch1.ogg', 35, 1)
 	if(iscarbon(hit_atom) && hit_atom != src)
 		var/mob/living/carbon/victim = hit_atom
 		if(victim.flying)
 			return
 		if(hurt)
-			victim.take_organ_damage(10)
-			take_organ_damage(10)
-			victim.Weaken(2 SECONDS)
-			Weaken(2 SECONDS)
+			victim.take_organ_damage(7.5 * speed)// speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
+			take_organ_damage(7.5 * speed)
```

Comment:
```suggestion
			var/damage = 7.5 * speed
			victim.take_organ_damage(damage) // speed while thrower is standing still is 2, while walking with an aggressive grab is 2.4
			take_organ_damage(damage)
```
no need to do the maths twice.

Reply:
I was trying some stuff and going way over the top with stuff like `log((10 + speed))` but damage = 10 + 1.5 * speed is way better, will implement both of these